merkle/simple_map.go

```diff
@@ -18,25 +18,25 @@ func NewSimpleMap() *SimpleMap {
 	}
 }
 
-func (sm *SimpleMap) Set(key string, value interface{}) {
+func (sm *SimpleMap) Set(key string, value Hasher) {
 	sm.sorted = false
 
-	// Is value Hashable?
-	var vBytes []byte
-	if hashable, ok := value.(Hashable); ok {
-		vBytes = hashable.Hash()
-	} else {
-		vBytes, _ = wire.MarshalBinary(value)
-	}
+	// Hash the key to blind it... why not?
+	khash := SimpleHashFromBytes([]byte(key))
```

Comment:
why not declare `key` as `Hasher` too and call `key.Hash()` here?

Reply:
I think most cases will just need a string or []byte, so optimizing for the typical case.